File: AlgorithmReview/Programmers/Lv2/카펫.swift
//
//  카펫.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2023/03/24.
//

import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    let sum = (brown + 4) / 2
    for i in 1...sum/2{
        let j = sum - i
        if((i-2) * (j-2) == yellow){
            return [j, i]
        }
    }
    return []
}

File: Python/카펫.py
def solution(brown, yellow):

    answer = []
    
    for x in range(1,2500):
        
        if yellow % x != 0:
            continue
            
        y = int(yellow/x)
        if x+y+2 == brown/2:
            answer = [x+2, y+2]
            break

    return sorted(answer, reverse=True)
